### ISSUE TITLE
Directories with trailing slash in .gitignore will not be exluded in version 6.17

### ISSUE THREAD

**Issue by hille721** [ansible/ansible-lint#3524]
##### Summary

directories with trailing slashes in `.gitignore` won't be exluded in ansible-lint 6.17.0.
I guess this is related to the reimplementation of the file exclusion logic (https://github.com/ansible/ansible-lint/pull/3507).

This statement https://github.com/ansible/ansible-lint/blob/98063af1d3b95475dc60e3eecbd9bec5c1683862/src/ansiblelint/file_utils.py#L541 will return `false` for a comparison of Path(`foo`) with `foo/` in the pathspec object. It will even return `false` for `Path('foo/')` as pathlib will strip trailing slashes (see https://github.com/python/cpython/issues/65238)

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.17.0 using ansible-core:2.14.6 ruamel-yaml:None ruamel-yaml-clib:None
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)
echo "foo/" > .gitignore
mkdir foo
echo 'foobar' > foo/bar.yml
ansible-lint -v
```

This will result in following error:

```
INFO     Loading ignores from .gitignore
WARNING  Listing 1 violation(s) that are fatal
load-failure: Failed to load or parse file.
foo/bar.yml:1


             Rule Violation Summary              
 count tag          profile rule associated tags 
     1 load-failure min     core, unskippable    

Failed after : 1 failure(s), 0 warning(s) on 2 files.
```

As you can see, it won't be excluded.

##### Desired Behavior

It should work the same as if the directory is specified without trailing slash:

```console (paste below)
echo "foo" > .gitignore
mkdir foo
echo 'foobar' > foo/bar.yml
ansible-lint -v
```

```
INFO     Loading ignores from .gitignore
INFO     Excluded: foo

Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
```

**Comment by hille721** [comment]
easy fix would be for example to change the lines https://github.com/ansible/ansible-lint/blob/98063af1d3b95475dc60e3eecbd9bec5c1683862/src/ansiblelint/file_utils.py#L563-L565

to 

``` python
pathspecs.append(
    pathspec.GitIgnoreSpec.from_lines(line.strip("/") for line in f.read().splitlines()),
)

**Comment by ssbarnea** [comment]
@hille721 Thanks for the very well written report! Would you mind making PR to address it?

**Comment by hille721** [comment]
`--exclude foo` and `--exclude foo/` also have a different behavior. Both are working but with `--exclude foo/` ansible-lint will look into `foo` directory first and then exclude all file individually. This could lead to a light performance impact on large directories. 

```
$ ansible-lint --exclude foo/ -v
INFO     Excluded: foo/bar.yml
INFO     Excluded: foo/bar2.yml

Passed with production profile: 0 failure(s), 0 warning(s) on 0 files.


$ ansible-lint --exclude foo -v
INFO     Excluded: foo

Passed with production profile: 0 failure(s), 0 warning(s) on 0 files.

**Comment by ssbarnea** [comment]
That is brilliant, I wanted to suggest you to look at black as they are heavy users of pathspec.
